tests/e2e/specs/shopper/product-query/product-query-with-templates.test.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { deleteAllTemplates } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	BASE_URL,
+	goToTemplateEditor,
+	saveTemplate,
+	useTheme,
+} from '../../../utils';
+import {
+	addProductQueryBlock,
+	block,
+	configurateProductQueryBlock,
+	getProductsNameFromClassicTemplate,
+	getProductsNameFromProductQuery,
+} from './utils';
+
+describe( `${ block.name } Block`, () => {
+	useTheme( 'emptytheme' );
+	afterAll( async () => {
+		await deleteAllTemplates( 'wp_template' );
+		await deleteAllTemplates( 'wp_template_part' );
+	} );
+	describe( 'with Product Catalog Template', () => {
+		beforeAll( async () => {
+			const productCatalogTemplateId =
+				'woocommerce/woocommerce//archive-product';
+
+			await goToTemplateEditor( {
+				postId: productCatalogTemplateId,
+			} );
+			await addProductQueryBlock();
+			await configurateProductQueryBlock();
+			await page.waitForNetworkIdle();
+			await saveTemplate();
+			await page.waitForNetworkIdle();
+			await page.goto( BASE_URL + '/shop', {
+				waitUntil: 'networkidle0',
+			} );
+		} );
+
+		it( 'should render the same products in the same position of the classic template', async () => {
+			const classicProducts = await getProductsNameFromClassicTemplate();
+			const products = await getProductsNameFromProductQuery();
+
+			expect(
+				classicProducts.every( ( val, idx ) => val === products[ idx ] )
+			).toBe( true );
```

Comment:
I would prefer if we compared the arrays directly:

```suggestion
			expect( classicProducts ).toEqual( products );
```

For this to work, however, we would need to move the `afterAll()` from L25 to inside each `describe()`. And then, that causes the error @albarin fixed in https://github.com/WordPress/gutenberg/pull/45992. Related issue: https://github.com/woocommerce/woocommerce-blocks/issues/7744.

So I think we can move ahead without making the change, but it would be good to note that in #7744 so we don't forget. How does it sound?